src/detect.c

```diff
@@ -1019,6 +1101,21 @@ DetectRunTxSortHelper(const void *a, const void *b)
         return s0->id > s1->id ? 1 : -1;
 }
 
+/** \internal
+ * \brief sort helper for sorting alerts by priority
+ *
+ * The id field is set from Signature::num, so we have higher priority (lower ids) first
+ */
+static int AlertQueueSortHelper(const void *a, const void *b)
```

Comment:
should maybe make sure that if they are equal we sort by PacketAlert::txid?

Reply:
okey dokey!